docs/specs/target.md

```diff
@@ -0,0 +1,155 @@
+  * MUST NOT deploy Vector aggregator nodes, since the Vector aggregator can be
+    configured to assume agent responsibilities.
+  * MUST deploy with Vector's [default agent configuration][default_agent_configuration]
+    which largely covers the agent architecture
+    [design recommendations][agent_architecture_design].
+* Sizing
+  * MUST deploy as a good infrastructure citizen, giving resource priority to
+    other services on the same node.
+  * SHOULD be limited to 2 vCPUs, MUST be overridable by the user.
+  * SHOULD be limited to 2 GiB of memory per vCPU (4 GiB in this case), MUST be
+    overridable by the user.
+  * SHOULD be limited to 1 GiB of disk space, MUST be overridable by the user.
+
+### 4. Aggregator Architecture
+
+* Architecture
+  * MUST deploy as a stateful service on dedicated nodes, Vector is the only
+    service on the node.
+  * MUST deploy with a persistent disk that is available between deployments.
+  * MUST deploy with Vector's [default aggregator configuration][default_aggregator_configuration]
+    which largely covers the aggregator architecture
+    [design recommendations][aggregator_architecture_design].
+    in order to achieve durability with disk buffers and source checkpoints.
+  * SHOULD deploy within one Cluster or VPC at a time.
+  * Configured Vector ports, including non-default user configured ports,
+    SHOULD be automatically accessible within the Cluster or VPC.
+  * Configured Vector sources, including non-default user configured sources,
+    SHOULD be automatically discoverable via target service discovery
+    mechanisms.
+* High Availability
+  * SHOULD deploy across 2 nodes, MUST be overridable by the user.
+  * SHOULD deploy across 2 availability zones, MUST be overridable by the user.
+* Sizing
+  * MUST deploy in a way that takes full advantage of all system resources.
+    The Vector service should not be artificially limited with resource
+    limiters such as cgroups.
+  * SHOULD request 8 vCPUs, MUST be overridable by the user.
+  * SHOULD request 2 GiB of memory per vCPU (16 GiB in this case), MUST be
+    overridable by the user.
+  * SHOULD request 6 GiB of disk space per vCPU (48 GiB in this case), MUST be
+    overridable by the user.
+  * SHOULD use the following instances if deployed in a listed cloud.
+    * AWS SHOULD default to `c6g.2xlarge` instances with 48 GiB of EBS `io2`
+      disk space.
+    * Azure SHOULD default to `f8` instances with 48 GiB of standard SSD disk
+      space.
+    * GCP SHOULD default to `c2` instances with 8 vCPUS, 16 GiB of memory, and
+      48 GiB of SSD persisted disk space.
+* Scaling
+  * Load-balancing?...
+  * Autoscaling SHOULD be enabled by default driven by an 85% CPU utilization
+    target over a rolling 5 minute window.
+  * Autoscaling SHOULD have a stabilization period of 5 minutes.
+
+### 5. Unified Architecture
+
+TODO: Should we support this as a top-level architecture, or have users deploy
```

Comment:
As a top-level arch: This would be awkward to accomplish with the current chart design and possibly with upgrade processes down the line. I'd lean towards two deployments.